lqm/week2/day2/pkg/worker/worker.go

```diff
@@ -0,0 +1,51 @@
+package worker
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+)
+
+type Result struct {
+	JobID string
+	State int // 0: Failed, 1: Success
+}
+
+type JobHandler func() Result
+
+type Job struct {
+	ID      string
+	Payload string
+	Handler JobHandler
+}
+
+const time_limit = 100 * time.Second
+
+func Worker(ctx context.Context, workerID int, jobs <-chan Job, results chan<- Result, wg *sync.WaitGroup) {
+	defer wg.Done()
+	timer := time.NewTimer(time_limit)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case job, ok := <-jobs:
+			if !ok {
+				// Job channels are closed
+				return
+			}
+			if !timer.Stop() {
+				<-timer.C
+			}
```

Comment:
```suggestion
			if !timer.Stop() {
				<-timer.C
			}
```
Could you explain the purpose of these lines?

Reply:
If the timer has fired before we stop the timer by `timer.Stop()`, `timer.Stop()` will return `false` and there will be a signal waiting in `timer.C`, I need to drain the `timer.C` channel to prevent it from being blocked

Reply:
I think line 40 will be blocked in your goroutine due to the update of Timer from go.1.23
https://pkg.go.dev/time#NewTimer

> I need to drain the timer.C channel to prevent it from being blocked

I don't get this point, could you explain more detail with a demo?